tests/test_derivatives.py

```diff
@@ -63,6 +69,40 @@ def test_stencil_derivative(self, SymbolType, dim):
         assert(np.allclose(u_di.data, 66.6))
         assert(np.allclose(u_dii.data, 66.6))
 
+    @pytest.mark.parametrize('SymbolType, derivative, dim, expected', [
+        (Function, ['dx2'], 3, 'Derivative(u(x, y, z), (x, 2))'),
+        (Function, ['dx2dy'], 3, 'Derivative(u(x, y, z), (x, 2), y)'),
+        (Function, ['dx2dydz'], 3, 'Derivative(u(x, y, z), (x, 2), y, z)'),
+        (Function, ['dx2', 'dy'], 3, 'Derivative(Derivative(u(x, y, z), (x, 2)), y)'),
+        (Function, ['dx2dy', 'dz2'], 3,
+         'Derivative(Derivative(u(x, y, z), (x, 2), y), (z, 2))'),
+        (TimeFunction, ['dx2'], 3, 'Derivative(u(t, x, y, z), (x, 2))'),
+        (TimeFunction, ['dx2dy'], 3, 'Derivative(u(t, x, y, z), (x, 2), y)'),
+        (TimeFunction, ['dx2', 'dy'], 3,
+         'Derivative(Derivative(u(t, x, y, z), (x, 2)), y)')
+    ])
+    def test_unevaluation(self, SymbolType, derivative, dim, expected):
```

Comment:
Can we add a single new case, in which **lots** of derivatives are taken, checking that unevaluation really does the right thing. Such as, ``.dx.dy.dx2.dz.dzdy``

Reply:
Done